### DISCUSSION TITLE
Applying Extension Methods to an Interface in C#

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#192] [Q&A] [answered]
Hello,

Could you please write an example for applying extension methods to an interface in C#? Which allows me to call the extension method on all the classes that implement the interface.

Thank you in advance

**Comment by christiannagel** [comment, marked as answer]
Hi Shervan,

Page 75 shows an extension method that extends the string type.
On page 230 you can see an extension method for `IEnumerable<TSource>`:

```csharp
public static IEnumerable<TSource> Where<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate> predicate)
{
    foreach (TSource item in source)
    {
        if (predicate(item))
            yield return item;
    }
}
```

With this, you can invoke the `Where` method on every object implementing `IEnumerable<T>`.

**Reply by ShervanN** [reply]
Great.
